parser/src/ast/declarations.rs

```diff
@@ -76,6 +78,56 @@ pub enum Declaration {
     IntegrityConstraints(Span<Vec<Statement>>),
 }
 
+/// Represents a bus declaration in an AirScript module.
+#[derive(Debug, Clone, Spanned)]
+pub struct Bus {
+    #[span]
+    pub span: SourceSpan,
+    pub name: Identifier,
+    pub bus_type: BusType,
+}
+impl Bus {
+    /// Creates a new bus declaration
+    pub const fn new(span: SourceSpan, name: Identifier, bus_type: BusType) -> Self {
+        Self {
+            span,
+            name,
+            bus_type,
+        }
+    }
+}
+#[derive(Default, Hash, Debug, Clone, PartialEq, Eq)]
+pub enum BusType {
+    /// A multiset bus
+    #[default]
+    Unit,
+    /// A logup bus
+    Mult,
```

Comment:
I find it a tad bit confusing that the "multiset" bus type is called `Unit`, while `Mult` refers to the LogUp-style bus. IIUC, `Mult` here refers to "multiplicative", in contrast to a non-multiplicative bus, hence "unit".

I think this is a case again where abbreviating things just obscures the meaning too much.